### DISCUSSION TITLE
Add SKU number from createListing

### DISCUSSION THREAD

**Discussion by adg-0** [etsy/open-api#1036] [Q&A] [answered]
If I understand correctly, there is no way to set a SKU number to a listing while creating it, right ?

The only way is to create the listing first, then update it with the SKU ?

**Comment by Gareth064** [comment, marked as answer]
I can't find it, but I'm almost certain you create the listing and then add inventory to it? And the inventory is what contains products and their skus

**Reply by adg-0** [reply]
Ok thank you, I was just making sure there wasn't a more direct way.

**Comment by emergingdzns** [comment]
@Gareth064 is correct. In V3 the skus are only added as part of inventory. Even if it's a single item with no variations. When you create the listing, the API will generate the inventory for the one product. You can then pull the inventory for that listing and use that inventory as sort of your template for either just changing the sku or for adding more variations. Just be sure to remove the fields that they don't want to get back. Those are detailed in the tutorials on the developers site [here](https://developers.etsy.com/documentation/tutorials/listings#updating-inventory).

Also do not ever send a partial inventory. It has to be the full inventory of variations (if you have more than the one product in the listing). Otherwise you'll lose all other variations.

**Reply by adg-0** [reply]
Yes, that's basically what I do. I do it right after createDraftListing
